waterlog/src/reducers/TankLevelReducer.js

```diff
@@ -12,18 +12,21 @@ const initialState = {
 export default function (state = initialState, action) {
     switch (action.type) {
         case FETCH_TANK_LEVELS_BEGIN:
+            console.log('begin test')
             return {
                 ...state,
                 loading: true,
                 error: null
             };
         case FETCH_TANK_LEVELS_SUCCESS:
+            console.log('levels', action.payload)
+            console.log('levels', action.payload.levels)
             return {
-                ...state,
-                loading: false,
-                items: action.payload.levels,
-                error: null
-            }
+                    ...state,
```

Comment:
Why are you returning the initial or default object with the modified object in your action creator?

Reply:
I set the initial state first. I am not really sure why it is being done like this. But of all the reducer examples I saw. They use the same approach.

Reply:
I'm pretty reluctant about this approach.
Knowing that the reducer returns the correct object based on the action type selected, I would suggest returning the default state when it fulfills its specific condition else return the new state.

Seeing it being done like that should be questioned when you don't understand it or don't approve of.

Reply:
I have been doing it the same way for more than one reducer and @clewarne has approved of how I was returning the object.

Reply:
@clewarne Help me to understand why @LisaQ04 is returning the initial or default object with the modified object in his reducer?

Reply:
@laurent-mbuyu what is happening here is that we don't want to lose any of the initial properties on the state. Not all properties always change so we first make sure to keep everything on the state and then only overriding the changed properties
```{
   ...state,
   // here override any properties that need to change
```